test/integration/statefulset/statefulset_test.go

```diff
@@ -719,17 +725,43 @@ func TestStatefulSetStartOrdinal(t *testing.T) {
 			}
 
 			var podNames []string
+			var podLabelOrdinals []string
 			for _, pod := range pods.Items {
 				podNames = append(podNames, pod.Name)
+				if idx, ok := pod.Labels[appsv1.PodIndexLabel]; !ok {
+					t.Errorf("Expected pod index label with key: %s", appsv1.PodIndexLabel)
+				} else {
+					podLabelOrdinals = append(podLabelOrdinals, idx)
+				}
 			}
 			ignoreOrder := cmpopts.SortSlices(func(a, b string) bool {
 				return a < b
 			})
+			ignoreOrderForOrdinals := cmpopts.SortSlices(func(a, b string) bool {
```

Comment:
Hm, this is verbose (repeated conversion for a and b), maybe just read all the indexes as int, and then sort ints.